File: wiringX/read.c
/*
	Copyright (c) 2016 CurlyMo <curlymoo1@gmail.com>

  This Source Code Form is subject to the terms of the Mozilla Public
  License, v. 2.0. If a copy of the MPL was not distributed with this
  file, You can obtain one at http://mozilla.org/MPL/2.0/.
*/

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>
#include <unistd.h>
#include "wiringx.h"

#define gpio 8

int main(int argc, char *argv[]) {

	if(wiringXSetup("rock4", NULL) == -1) {
		wiringXGC();
		return -1;
	}
	pinMode(gpio, PINMODE_INPUT);

	while(1) {
		printf("Reading from GPIO %d: value is %d\n", gpio, digitalRead(gpio));
		sleep(1);
	}

	return 0;
}

File: wiringX/python/read.py
#!/usr/bin/env python3
# -*- encoding: utf-8 -*-

import os
import sys
from time import sleep
from wiringX import gpio

gpio.setup(gpio.ROCK4);

gpio.pinMode(gpio.PIN8, gpio.PINMODE_INPUT);

try:
	while True:
		print("Reading from pin 8: "+str(gpio.digitalRead(gpio.PIN8)));
		sleep(1);
		print("Reading from pin 8: "+str(gpio.digitalRead(gpio.PIN8)));
		sleep(1);
except KeyboardInterrupt:
	pass
